cmd/flipt/main.go

```diff
@@ -172,11 +172,14 @@ func determinePath(cfgPath string) string {
 	}
 
 	_, err := os.Stat(fliptConfigFile)
-	if !errors.Is(err, fs.ErrNotExist) {
-		defaultLogger.Warn("unexpected error checking configuration path", zap.String("config_path", fliptConfigFile), zap.Error(err))
+	if err != nil {
+		if !errors.Is(err, fs.ErrNotExist) {
+			defaultLogger.Warn("unexpected error checking configuration path", zap.String("config_path", fliptConfigFile), zap.Error(err))
+		}
+		return defaultCfgPath
```

Comment:
You will need to delete the next few lines too, but this might be a little easier to grok:
```suggestion
	if err == nil {
		return fliptConfigFile
	}
	if !errors.Is(err, fs.ErrNotExist) {
		defaultLogger.Warn("unexpected error checking configuration path", zap.String("config_path", fliptConfigFile), zap.Error(err))
	}
	
	return defaultCfgPath
```